example/lib/main.dart

```diff
@@ -0,0 +1,258 @@
+            borderColor: theme.messageReactionBackGroundColor,
+          ),
+          imageMessageConfig: ImageMessageConfiguration(
+            onTap: (imageUrl) => print("On Image Tap $imageUrl"),
+            shareIconConfig: ShareIconConfiguration(
+              defaultIconBackgroundColor: theme.shareIconBackgroundColor,
+              defaultIconColor: theme.shareIconColor,
+              onPressed: (imageUrl) => Fluttertoast.showToast(
+                msg: "Add custom share widget",
+                toastLength: Toast.LENGTH_SHORT,
+                gravity: ToastGravity.BOTTOM,
+                timeInSecForIosWeb: 1,
+                backgroundColor: theme.outgoingChatBubbleColor,
+                textColor: Colors.white,
+                fontSize: 16.0,
+              ),
+            ),
+          ),
+        ),
+        showReceiverProfileCircle: true,
+        profileCircleConfig:
+            ProfileCircleConfiguration(profileImageUrl: Data.profileImage),
+        repliedMessageConfig: RepliedMessageConfiguration(
+          backgroundColor: theme.repliedMessageColor,
+          verticalBarColor: theme.verticalBarColor,
+          textStyle: const TextStyle(
+            color: Colors.white,
+            fontWeight: FontWeight.bold,
+            letterSpacing: 0.25,
+          ),
+          replyTitleTextStyle: TextStyle(color: theme.repliedTitleTextColor),
+        ),
+        swipeToReplyConfig: SwipeToReplyConfiguration(
+          replyIconColor: theme.swipeToReplyIconColor,
+          onLeftSwipe: (message, sendBy) =>
+              print("Message $message SendBy $sendBy"),
+          onRightSwipe: (message, sendBy) =>
+              print("Message $message SendBy $sendBy"),
+        ),
+      ),
+    );
+  }
+
+  void _onSendTap(String message, ReplyMessage replyMessage) {
+    final id = int.parse(Data.messageList.last.id) + 1;
+    if (replyMessage.message.isNotEmpty) {
+      setState(() {
+        Data.messageList.add(
+          Message(
+            id: id.toString(),
+            createdAt: DateTime.now(),
+            message: message,
+            sendBy: sender.id,
+            replyMessage: replyMessage,
+          ),
+        );
+      });
+    } else {
+      setState(() {
+        Data.messageList.add(
+          Message(
+            id: id.toString(),
+            createdAt: DateTime.now(),
+            message: message,
+            sendBy: sender.id,
+          ),
+        );
+      });
+    }
+  }
+
+  void _onPressed() {
+    setState(() {
+      if (isDarkTheme) {
+        theme = LightTheme();
+        isDarkTheme = false;
+      } else {
+        theme = DarkTheme();
+        isDarkTheme = true;
+      }
+    });
+  }
+
+  void _setReaction(String emoji, String messageId) {
+    final message =
+        Data.messageList.firstWhere((element) => element.id == messageId);
+    final index = Data.messageList.indexOf(message);
+    setState(() {
```

Comment:
set state should not used by user all this needs to be handled inside our package